src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -317,17 +312,26 @@ void RNN<OutputLayerType, InitializationRuleType>::Gradient(
     else
     {
       outputLayer.Backward(std::move(boost::apply_visitor(
-          outputParameterVisitor, network.back())), std::move(currentTarget),
+          outputParameterVisitor, network.back())),
+          std::move(responses.submat((rho - seqNum - 1) * targetSize, begin,
+          (rho - seqNum) * targetSize - 1, begin + batchSize - 1)),
           std::move(error));
     }
 
     Backward();
-    Gradient();
+    Gradient(std::move(predictors.submat((rho - seqNum - 1) * inputSize, begin,
+        (rho - seqNum) * inputSize - 1, begin + batchSize - 1)));
     gradient += currentGradient;
   }
```

Comment:
I missed ``gradient /= batchSize;``, do you like to add that step or should I commit the changes to my branch?

Reply:
@zoq Done!